public/app/features/dashboard/components/GenAI/GenAIPanelDescriptionButton.tsx

```diff
@@ -24,7 +24,9 @@ const DESCRIPTION_GENERATION_STANDARD_PROMPT =
   `The description should be, at most, ${PANEL_DESCRIPTION_CHAR_LIMIT} characters.`;
 
 export const GenAIPanelDescriptionButton = ({ onGenerate, panel }: GenAIPanelDescriptionButtonProps) => {
-  const messages = React.useMemo(() => getMessages(panel), [panel]);
+  // We want to recompute messages when panel type changes
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const messages = React.useMemo(() => getMessages(panel), [panel, panel.type]);
```

Comment:
We can remove the memo and re-compute `getMessages` for every click. This should fix this issue.